internal/webserver/server.go

```diff
@@ -173,9 +180,75 @@ func (webserver *WebServer) versionHandler(writer http.ResponseWriter, _ *http.R
 	return
 }
 
+// swagger:operation POST /secret Secret
+//
+// Secret
+//
+// Posts the a secret to the SDK that the SDK will add to the secret client.
+//
+// ---
+// produces:
+// - application/json
+//
+// Schemes:
+//  - http
+//
+// Responses:
+//  '201':
+//    description: Posted a secret to the SDK
+//    schema:
+//      "$ref": "#/definitions/Secret"
+//   '400':
+//    description: Bad request could not decode JSON payload
+//    schema:
+//      "$ref": "#/definitions/Secret"
+//
+func (webserver *WebServer) secretHandler(writer http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(req.Body)
+	var secretData SecretData
+
+	err := decoder.Decode(&secretData)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to parse json payload: %v", err)
+		webserver.writeResponse(writer, msg, http.StatusBadRequest)
+		return
+	}
+
+	if len(secretData.Secrets) == 0 {
+		webserver.writeResponse(writer, "Missing required field 'Secrets'", http.StatusBadRequest)
+		return
+	}
+
```

Comment:
Any validation on Path for valid characters, i.e. $, @, ! etc will cause error?

Reply:
good catch: try some path like this: `%$xxx` may throw off the api call....